code/controllers/subsystem/admin_verbs.dm

```diff
@@ -0,0 +1,197 @@
+	context_map = SSadmin_verbs.context_map
+	admin_linkup_map = SSadmin_verbs.admin_linkup_map
+	assosciations_by_ckey = SSadmin_verbs.assosciations_by_ckey
+
+/datum/controller/subsystem/admin_verbs/Initialize()
+	RegisterSignal(src, COMSIG_SUBSYSTEM_POST_INITIALIZE, PROC_REF(assosciate_with_waiting))
+	admin_verb_map = list()
+	admin_linkup_map = list()
+	generate_holder_map()
+	context_map = list()
+	populate_context_map(context_map)
+	assosciations_by_ckey = list()
+	return SS_INIT_SUCCESS
+
+/datum/controller/subsystem/admin_verbs/proc/generate_stat_data(client/target)
+	var/static/list/abbreviations = list(
+		"ERT"
+	)
+	var/static/list/cached_formats = list()
+
+	if(!initialized || !target.holder)
+		return list()
+
+	var/list/stat_data = list()
+	for(var/verb_type in assosciations_by_ckey[target.ckey])
+		var/list/verb_information = admin_verb_map[verb_type]
+		var/verb_permissions = verb_information[VERB_MAP_PERMISSIONS]
+		if(!check_rights_for(target, verb_permissions))
+			continue
+
+		var/verb_module = lowertext(verb_information[VERB_MAP_MODULE])
+		if(!verb_module || verb_module == "null")
+			continue
+
+		if(!cached_formats[verb_module])
+			var/verb_module_formatted = ""
+			for(var/verb_module_part in splittext(verb_module, "_"))
+				if(verb_module_part in abbreviations)
+					verb_module_formatted += "[uppertext(verb_module_part)] "
+				else
+					verb_module_formatted += "[capitalize(verb_module_part)] "
+			verb_module_formatted = copytext(verb_module_formatted, 1, -1)
+			cached_formats[verb_module] = verb_module_formatted
+
+		var/original_name = verb_information[VERB_MAP_NAME]
+		if(!cached_formats[original_name])
+			var/formatted_name = ""
+			for(var/name_part in splittext(original_name, "_"))
+				if(name_part in abbreviations)
+					formatted_name += "[uppertext(name_part)] "
+				else
+					formatted_name += "[capitalize(name_part)] "
+			formatted_name = copytext(formatted_name, 1, -1)
+			cached_formats[original_name] = formatted_name
+
+		var/verb_desc = verb_information[VERB_MAP_DESCRIPTION]
+		if(!stat_data[cached_formats[verb_module]])
+			stat_data[cached_formats[verb_module]] = list()
+		stat_data[cached_formats[verb_module]] += list(list(cached_formats[original_name], verb_desc, original_name))
+	var/sorted_stat_data = list()
+	for(var/verb_category in stat_data)
+		sorted_stat_data[verb_category] = sort_list(stat_data[verb_category], GLOBAL_PROC_REF(cmp_admin_verb_name))
+	return sorted_stat_data
+
+/proc/cmp_admin_verb_name(list/info_left, list/info_right)
+	return sorttext(info_right[1], info_left[1])
+
+/datum/controller/subsystem/admin_verbs/proc/populate_context_map(list/context_map)
+	return
+
+/datum/controller/subsystem/admin_verbs/proc/generate_holder_map()
+	admin_verb_map = list()
+	holder_map = list()
+	var/list/processing = typecacheof(sort_list(subtypesof(/mob/admin_module_holder), GLOBAL_PROC_REF(cmp_typepaths_asc)))
+	processing -= typecache_next_level(/mob/admin_module_holder)
```

Comment:
what is going on here

Reply:
When I go to assemble verb information into a stat data list for transmission to the statpanel, I call dynamic_map_generate on the holder to create information.
The subtype which is a module flag, aka the first level descendant, doesn't have an override for the proc and instead returns null; which causes an error during stat information generation because I index into that list. I could probably do something like module_verbname but I don't believe macro replacements work with stuff like ##module_##verb_name